tests/FunctionalTestCase.php

```diff
@@ -382,9 +384,16 @@ protected function skipIfTransactionsAreNotSupported()
             $this->markTestSkipped('Transactions are not supported on standalone servers');
         }
 
-        // TODO: MongoDB 4.2 should support sharded clusters (see: PHPLIB-374)
         if ($this->isShardedCluster()) {
-            $this->markTestSkipped('Transactions are not supported on sharded clusters');
+            if (!$this->isShardedClusterUsingReplicasets()) {
```

Comment:
We can't easily figure out what storage engine is in use on sharded clusters, so we're skipping that check when running a sharded setup. I don't think we even run any tests on the MMAPv1 storage engine at all in PHPLIB. This may change once we move library tests to Evergreen, but I'd revisit this then and figure out a more robust solution.

Reply:
MMAPv1 was completely [removed in 4.2](https://docs.mongodb.com/master/release-notes/4.2/#removed-mmapv1-storage-engine), so this shouldn't be a concern.